Topics/Development_Process/GraphQL_VS_REST.md

```diff
@@ -0,0 +1,94 @@
+# GraphQL and REST APIs
+
+## Table of Contents
+1. [Overview & Motivation](#overview--motivation)
+    - [What is an API?](#what-is-an-api)
+    - [Why do we need APIs?](#why-do-we-need-apis)
+    - [Designing APIs](#designing-apis)
+2. [REST Overview](#rest-overview)
+3. [GraphQL Overview](#graphql-overview)
+4. [When does REST make sense to use?](#when-does-rest-make-sense-to-use)
+5. [When does GraphQL make sense to use?](#when-does-graphql-make-sense-to-use)
+6. [Quick Reference](#quick-reference-for-common-uses)
+7. [Summary](#summary)
+8. [Additional Readings](#additional-reading)
+
+## Overview & Motivation
+### What is an API?
+An API is an **application programming interface**, or in simpler words, a way for two pieces of software to communicate with one another.
+
+On this page, we will refer to things called `endpoints`. These are basically access ways for our API - you use these endpoints to communicate with the software.
+
+### Why do we need APIs?
+Without APIs, there would be no way for software to communicate between one another. In fact, even within the same project, different components may have APIs between them to facilitate communication. Without APIs, software would never be able to interact with anything. Even something as simple as loading Google's webpage deals with many dozens of APIs without you knowing! 
+
+### Designing APIs
+In this article, we will specifically discuss APIs that deal with sending requests over the internet, which is a very special case of APIs. We will cover two special commonly used types of APIs:
+1. REST
+2. GraphQL 
+
+On this page, you will learn about the differences between the two, and hopefully be able to make a confident decision on which one you need to use, based on your project requirements. 
+
+## REST Overview
+A REST API is any API that follows the REST architecture. The implementation of it can be handled in many pre-existing libraries, so this page will focus on the important details:
+1. In REST, you have to specify the `type` of the request you are sending when you call an API endpoint.
+    + Some common examples include `GET`, `POST`, `PUT`, and `DELETE`. Others exist, but these are the ones used most often.
+2. REST is not specific; when you call an endpoint, you will get *all* of the data that endpoint can give, whether you like it or not.
+    + This can be ideal when you require a lot of data, but can also lead to larger than necessary load on your system if you have to call many API endpoints in quick succession.
+3. Every difference type of request in REST has to be called to a different endpoint. For instance, if you want to `GET` all `Users` and `Cars` at the same time, with a typical REST API, you would need to send two seprate `GET` requests to a `/users` and `/cars` endpoint.

```

Comment:
```suggestion
3. Every different type of request in REST has to be called to a different endpoint. For instance, if you want to `GET` all `Users` and `Cars` at the same time, with a typical REST API, you would need to send two separate `GET` requests to a `/users` and `/cars` endpoint.

```

Reply:
Oops 😅 . That's really embarrassing for me. Thanks for the catch :)